src/parser/parser_verification.js

```diff
@@ -0,0 +1,150 @@
+import { requiredValues } from '../util/requiredValues';
+import { parserUtils } from './parser_utils';
+
+/**
+ * Verify and trigger warnings if a node required value is not set.
+ * @param  {Node} node - The node element.
+ * @param  {Function} emit - Emit function used to trigger Warning event.
+ * @param  {Boolean} [isAdInline] - True if node is contained inside a InLine node.
+ *
+ */
+function verifyRequiredValues(node, emit, isAdInline) {
+  if (!node || !node.nodeName) {
+    return;
+  }
+  if (node.nodeName === 'InLine') {
+    isAdInline = true;
```

Comment:
why are we passing `isAdInline` as a param to the function if you're checking for it here?

Reply:
`isAdInline` is used to know if a node is contained inside an InLine. Since it's a recursive function, it can go through multiple childs levels. I need to pass it through recursive iteration to know when a resource requires it ( one of StaticResource, IFrameResource, HTMLResource is required for the nodes NonLinear, Companion, or Icon when InLine format is used). As discussed in person I'll explain it in the JSDoc